Adapter/Aggregation/Builder.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Algolia\AlgoliaSearchElastic\Adapter\Aggregation;
+
+use Algolia\AlgoliaSearch\Model\Indexer\Product;
+use Algolia\AlgoliaSearchElastic\Helper\ElasticAdapterHelper;
+use Magento\Catalog\Model\ProductFactory;
+use Magento\Elasticsearch\SearchAdapter\Aggregation\Builder as ElasticSearchBuilder;
+use Magento\Elasticsearch\SearchAdapter\Aggregation\Builder\BucketBuilderInterface;
+use Magento\Elasticsearch\SearchAdapter\Aggregation\DataProviderFactory;
+use Magento\Elasticsearch\SearchAdapter\QueryContainer;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Search\Dynamic\DataProviderInterface;
+use Magento\Framework\Search\RequestInterface;
+
+class Builder extends ElasticSearchBuilder
```

Comment:
Does it need to extend ES Builder at the end ?

Reply:
At this point, I don't think so! I extended it at first as to not recreate the variables. I think it would be okay to integrate them and have our builder standalone.